Cython/Compiler/ModuleNode.py

```diff
@@ -1699,6 +1686,31 @@ def generate_dealloc_function(self, scope, code):
         if cdealloc_func_entry is None:
             code.putln("#endif")
 
+    def generate_finalizer_call(self, needs_gc, code):
+        code.putln("#if CYTHON_USE_TP_FINALIZE")
+
+        if needs_gc:
+            finalised_check = '!_PyGC_FINALIZED(o)'
+        else:
+            finalised_check = (
+                '(!PyType_IS_GC(Py_TYPE(o)) || !_PyGC_FINALIZED(o))')
+        code.putln(
+            "if (unlikely("
+            "(PY_VERSION_HEX >= 0x03080000 || __Pyx_PyType_HasFeature(Py_TYPE(o), Py_TPFLAGS_HAVE_FINALIZE))"
+            " && Py_TYPE(o)->tp_finalize) && %s) {" % finalised_check)
+        # if instance was resurrected by finaliser, return
+
+        code.putln("{")
+        code.putln("PyObject *etype, *eval, *etb;")
+        code.putln("PyErr_Fetch(&etype, &eval, &etb);")
+        code.putln("const int ressurected = PyObject_CallFinalizerFromDealloc(o);")
+        code.putln("PyErr_Restore(etype, eval, etb);")
+        code.putln("if (ressurected) return;")
+        code.putln("}")
```

Comment:
C89. And spelling :). And we don't need the extra braces since all of this is inside an `if … {}`
```suggestion
        code.putln("int resurrected;")
        code.putln("PyObject *etype, *eval, *etb;")
        code.putln("PyErr_Fetch(&etype, &eval, &etb);")
        code.putln("resurrected = PyObject_CallFinalizerFromDealloc(o);")
        code.putln("PyErr_Restore(etype, eval, etb);")
        # if instance was resurrected by finaliser, abort finalisation
        code.putln("if (resurrected) return;")
```